### PR TITLE
[wgsl] remove `outerProduct`

### PR DESCRIPTION
**Pull request by teoxoy**
This function has been removed from the WGSL spec (see https://github.com/gpuweb/gpuweb/pull/1290).

### CODE REVIEW

File: src/back/wgsl/writer.rs [hunk 1]
```diff
@@ -1568,7 +1567,7 @@ impl<W: Write> Writer<W> {
                     Mf::Unpack2x16snorm => Function::Regular("unpack2x16snorm"),
                     Mf::Unpack2x16unorm => Function::Regular("unpack2x16unorm"),
                     Mf::Unpack2x16float => Function::Regular("unpack2x16float"),
-                    Mf::Inverse => {
+                    Mf::Inverse | Mf::Outer => {
```
Comment (ErichDonGubler) [inline src/back/wgsl/writer.rs:1570]: question: Is there a reason we don't remove `Inverse` and `Outer` from `MathFunction` altogether? I'm assuming it's "only" for the benefit of this diagnostic here, and the fact that other front ends might emit these.

File: tests/in/glsl/math-functions.frag [hunk 2]
```diff
@@ -44,7 +44,8 @@ void main() {
     vec4 minOut = min(a, b);
     vec4 reflectOut = reflect(a, b);
     vec3 crossOut = cross(a.xyz, b.xyz);
-    mat4 outerProductOut = outerProduct(a, b);
+    // TODO: support outerProduct function in wgsl output
```
Comment (ErichDonGubler) [inline tests/in/glsl/math-functions.frag:47]: question: Would this `TODO`'d support mean we need to write a polyfill?

**Review by ErichDonGubler — APPROVED** [review]

File: tests/in/glsl/math-functions.frag (hunk 2 above)
Comment (teoxoy) [reply tests/in/glsl/math-functions.frag:47]: Yeah, similar to the inverse fn above.

File: src/back/wgsl/writer.rs (hunk 1 above)
Comment (teoxoy) [reply src/back/wgsl/writer.rs:1570]: > I'm assuming it's "only" for the benefit of this diagnostic here, and the fact that other front ends might emit these.

Yeah, GLSL and SPIR-V have those so, ideally we'd polyfill them.

**Pull request merged**